src/pages/todoSetup/hooks/useTodoSetup.js

```diff
@@ -0,0 +1,37 @@
+// Hooks
+import {useState } from 'react';
+
+function useTodoSetup(todoList, addTodoList) {
+    const [todoText, setTodoText] = useState("");
+
+       const onInputChange = (e) => {
+            setTodoText(e.target.value);
+        }
+    
+        const onFormSubmit = (e) => {
+            e.preventDefault();
+            if (!todoText) return;
+            
+            const newTodoInfo = {};
+            newTodoInfo.id = todoList.length + 1;
+            newTodoInfo.text = todoText;
+            newTodoInfo.isCompleted = false;
+    
+            const newTodoList = [...todoList, newTodoInfo];
+            addTodoList(newTodoList);
```

Comment:
Here we are spreading the todoList  and adding new todo item and then dispatching the update action.
Instead we can send the new todo only to addTodoList <> updateTodoList(newTodoInfo) 
Then how will you get the todoList, if you remember we discussed that we get dispatch and getState inside the action function.
So any time getState provides you the current latest state of the app, we can leverage same inside updateTodoList.

// const todoList = getState().todoList;